tests/unit/meta/Intersection.ts

```diff
@@ -0,0 +1,117 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import { WidgetBase } from '../../../src/WidgetBase';
+import { v } from '../../../src/d';
+import { ProjectorMixin } from '../../../src/main';
+import Intersection from '../../../src/meta/Intersection';
+import { stub } from 'sinon';
+import global from '@dojo/core/global';
+import { WidgetProperties } from '../../../src/interfaces';
+
+let rAF: any;
+let observerCallback: any;
+let intersectionObserver: any;
+
+function resolveRAF() {
+	for (let i = 0; i < rAF.callCount; i++) {
+		rAF.getCall(0).args[0]();
+	}
+	rAF.reset();
+}
+
+registerSuite({
+	name: 'meta - Intersection Observer',
+
+	beforeEach() {
+		rAF = stub(global, 'requestAnimationFrame');
+		intersectionObserver = stub(global, 'IntersectionObserver', function (callback: any) {
+			observerCallback = callback;
+			return {
+				observe: stub()
+			};
+		});
+	},
+
+	afterEach() {
+		rAF.restore();
+		intersectionObserver.restore();
+	},
+
+	'intersections cause invalidations'() {
+		const visibles: boolean[] = [];
+
+		class TestWidget extends ProjectorMixin(WidgetBase)<WidgetProperties> {
+			render() {
+				visibles.push(this.meta(Intersection).get('root') > 0);
+				return v('div', {
+					key: 'root'
+				});
+			}
+		}
+
+		const div = document.createElement('div');
+
+		document.body.appendChild(div);
+
+		const widget = new TestWidget();
+		widget.append(div);
+
+		resolveRAF();
+
+		const node = (<any> widget)._nodeMap.get('root');
```

Comment:
maybe we could just get this node out of the dom via a selector/id, then we wouldn't have to access internals?